akka-docs/src/test/java/jdocs/tutorial_1/ActorHierarchyExperiments.java

```diff
@@ -75,7 +75,7 @@ public Receive createReceive() {
 
 //#supervise
 class SupervisingActor extends AbstractActor {
-  ActorRef child = getContext().actorOf(Props.create(SupervisedActor.class), "supervised-actor");
+  ActorRef child = getContext().actorOf(Props.create(() -> new SupervisedActor()), "supervised-actor");
```

Comment:
I agree that `new` is better, but then we should use the static `SupervisedActor.props()` factory method convention instead. It's not only about serialization but also that the factory runs in a separate thread and this introduces the risk of accessing something that is not thread-safe from the enclosing scope (parent actor), and that is solved by the static factory method.

Reply:
Okey, will refactor to those after done with persistence

Reply:
Also a better Java convention is to use method references when possible (`SupervisedActor::new` or `SupervisedActor::props`).